program/src/instruction.rs

```diff
@@ -1,184 +1,373 @@
+        2,
+        writable,
+        name = "destination",
+        desc = "Destination account for withdrawn lamports"
+    )]
+    #[account(
+        3,
+        signer,
+        name = "stake_authority",
+        desc = "Stake authority"
+    )]
     HarvestStakeRewards,
 
     /// Slashes a stake account for the given amount
-    ///
-    /// Burns the given amount of tokens from the vault account, and reduces the
-    /// amount in the stake account.
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[w]` Stake account
-    /// 2. `[s]` Slash authority
-    /// 3. `[w]` Vault token account
-    /// 4. `[]` Vault authority, PDA with seeds `['token-owner', stake_config]`
-    /// 5. `[]` SPL Token program
-    ///
-    /// Instruction data: amount of tokens to slash
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        signer,
+        name = "slash_authority",
+        desc = "Stake account"
+    )]
+    #[account(
+        3,
+        writable,
+        name = "vault_token",
+        desc = "Vault token account"
+    )]
+    #[account(
+        4,
+        name = "vault_authority",
+        desc = "Vault authority"
+    )]
+    #[account(
+        5,
+        name = "spl_token_program",
+        desc = "SPL Token program"
+    )]
     Slash(u64),
 
     /// Sets new authority on a config or stake account
-    ///
-    /// 0. `[w]` Config or stake account
-    /// 1. `[s]` Current authority
-    /// 2. `[]` New authority
-    SetAuthority(Authority),
+    #[account(
+        0,
+        writable,
+        name = "account",
+        desc = "Config or Stake config account"
+    )]
+    #[account(
+        1,
+        signer,
+        name = "authority",
+        desc = "Current authority on the account"
+    )]
+    #[account(
+        2,
+        name = "new_authority",
+        desc = "Authority to set"
+    )]
+    SetAuthority(AuthorityType),
 
     /// Updates configuration parameters
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[s]` Config authority
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        signer,
+        name = "config_authority",
+        desc = "Stake config authority"
+    )]
     UpdateConfig(ConfigField),
 
     /// Moves SOL rewards to the config and updates the stake rewards total
-    ///
-    /// Accounts expected by this instruction:
-    ///
-    /// 0. `[w,s]` Reward payer
-    /// 1. `[w]` Config account
-    /// 2. `[]` System Program
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        signer,
+        name = "payer",
+        desc = "Reward payer"
```

Comment:
Also flipped.

Reply:
Same as previous, in this case `payer` stays "closer" to `system_program`.

Reply:
Reverted back to the original order.